Couple minor optimisations in Kernel

michalmuskala opened the pull request.

josevalim commented:
@michalmuskala please take a look at the build error.

josevalim commented on lib/elixir/lib/kernel.ex:
```diff
@@ -4573,7 +4576,7 @@ defmodule Kernel do
 
   defp split_words(string, [mod])
-  when mod == ?s or mod == ?a or mod == ?c do
+       when mod === ?s or mod === ?a or mod === ?c do
```
Let's revert this one. The style inconsistency with the codebase is not worth the very small gains.

michalmuskala commented:
Updated

josevalim commented:
@michalmuskala please check the build. :)

michalmuskala commented:
@josevalim with the failure, I think moving to the map syntax will produce a significantly inferior message, so we should skip it and keep `Map.replace!` as it is right now.

josevalim commented:
@michalmuskala sounds good to me!

michalmuskala commented:
I reverted to use `Map.replace!` where it makes a difference in the error message.

I also changed what's the `__struct__/1` implementation in case there's no `@enforce_keys` set. We don't need to track it.

josevalim commented:
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
